### PR TITLE
[RFR] Added type hints for browser module

### PR DESCRIPTION
**Pull request by quarckster**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -31,7 +31,14 @@ repos:
...
+    rev: v2.7.4
     hooks:
       - id: pyupgrade
         language_version: python3
+  - repo: git://github.com/luismayta/pre-commit-mypy
+    rev: 0.1.1
+    hooks:
+      - id: mypy
+        args:
+          - --ignore-missing-imports
```
Comment (RonnyPfannschmidt) [inline .pre-commit-config.yaml:43]: that one i'd like to see to gradually go away

ps, can we put all mypy config into setup.cfg?

File: src/widgetastic/browser.py [hunk 2]
```diff
@@ -28,6 +38,12 @@
...
 from .utils import retry_stale_element
 from .xpath import normalize_space
+from widgetastic.types import DefaultPluginType
+from widgetastic.types import ElementParent
+from widgetastic.types import LocatorAlias
+
+if TYPE_CHECKING:
+    from .widget.base import Widget
 
 Size = namedtuple("Size", ["width", "height"])
```
Comment (RonnyPfannschmidt) [inline src/widgetastic/browser.py:48]: i believe starting with python 3.6.1 named-tuple from typing can be used to type those as well

File: .pre-commit-config.yaml (hunk 1 above)
Comment (quarckster) [reply .pre-commit-config.yaml]: Done

File: src/widgetastic/browser.py (hunk 2 above)
Comment (quarckster) [reply src/widgetastic/browser.py]: Done

**Pull request merged**